src/main/java/org/example/springproject/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,108 @@
+package org.example.springproject.service.impl;
+
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.example.springproject.dto.order.OrderDto;
+import org.example.springproject.dto.order.OrderItemDto;
+import org.example.springproject.dto.order.PlaceOrderRequestDto;
+import org.example.springproject.dto.order.StatusUpdateRequestDto;
+import org.example.springproject.mapper.OrderMapper;
+import org.example.springproject.model.Order;
+import org.example.springproject.model.OrderItem;
+import org.example.springproject.model.ShoppingCart;
+import org.example.springproject.model.Status;
+import org.example.springproject.repository.cart.ShoppingCartRepository;
+import org.example.springproject.repository.order.OrderItemRepository;
+import org.example.springproject.repository.order.OrderRepository;
+import org.example.springproject.service.OrderService;
+import org.example.springproject.service.ShoppingCartService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartService shoppingCartService;
+
+    @Override
+    public List<OrderDto> getOrders(Long userId) {
+        List<Order> orders = orderRepository.findByUserId(userId);
+        return orders.stream().map(orderMapper::toDto).toList();
+    }
+
+    @Override
+    @Transactional
+    public OrderDto placeOrder(Long userId, PlaceOrderRequestDto requestDto) {
+        ShoppingCart cart = shoppingCartRepository.findByUserId(userId);
+        if (cart == null || cart.getCartItems().isEmpty()) {
+            throw new IllegalStateException("Cart is empty or not available");
+        }
+        Order order = createOrderFromCart(cart);
+        order.setShippingAddress(requestDto.getShippingAddress());
+        orderRepository.save(order);
+        shoppingCartService.clearCart(userId);
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public List<OrderItemDto> getOrderItems(Long userId, Long orderId) {
+        Order order = orderRepository.findByIdAndUserId(orderId, userId)
+                .orElseThrow(() -> new NoSuchElementException("Order with " + orderId
```

Comment:
```suggestion
                .orElseThrow(() -> new EntityNotFoundException("Order with " + orderId
```

check other places and fix